pkgs/by-name/ni/nixos-rebuild-ng/package.nix

```diff
@@ -53,22 +56,55 @@ python3.pkgs.buildPythonApplication {
       mv $out/bin/nixos-rebuild $out/bin/nixos-rebuild-ng
     '';
 
-  nativeCheckInputs = with python3.pkgs; [
+  nativeCheckInputs = with python3Packages; [
     pytestCheckHook
-    mypy
-    ruff
   ];
 
   pytestFlagsArray = [ "-vv" ];
 
-  postCheck = ''
-    echo -e "\x1b[32m## run mypy\x1b[0m"
-    mypy nixos_rebuild tests
-    echo -e "\x1b[32m## run ruff\x1b[0m"
-    ruff check nixos_rebuild tests
-    echo -e "\x1b[32m## run ruff format\x1b[0m"
-    ruff format --check nixos_rebuild tests
-  '';
+  # NOTE: this is a CI test rather than a build-time test because we want to
+  # keep the build closures small
+  passthru =
+    let
+      python-with-pkgs = python3.withPackages (
+        ps: with ps; [
+          mypy
+          pytest
+          ruff
+          types-tabulate
+          # dependencies
+          tabulate
+        ]
+      );
+    in
+    {
+      devShell = mkShell {
+        packages = [ python-with-pkgs ];
+        shellHook = ''
+          cd pkgs/by-name/ni/nixos-rebuild-ng/src || true
+        '';
+      };
+      tests.ci =
+        runCommand "${pname}-ci"
+          {
+            nativeBuildInputs = [
+              nixos-rebuild-ng # to trigger build
```

Comment:
I am not sure if this is the right way to do this. I don't think I have ever seen something like this in Nixpkgs before. While this achieves your goal, I think it's not what build inputs are intended for.

Reply:
Ok, I will remove it. I mostly added to make it easier to test locally (e.g.: I can run both the tests and CI with one command), but I can always run them separately and OfBorg will run the build automatically anyway so it doesn't make much difference.

Reply:
Yes, definitely a good idea.